src/main/java/com/redhat/devtools/lsp4ij/features/foldingRange/LSPFoldingRangeBuilder.java

```diff
@@ -67,12 +69,15 @@ protected void buildLanguageFoldRegions(@NotNull List<FoldingDescriptor> descrip
             for (FoldingRange foldingRange : foldingRanges) {
                 TextRange textRange = getTextRange(foldingRange, file, document);
                 if ((textRange != null) && (textRange.getLength() > 0)) {
-                    String collapsedText = foldingRange.getCollapsedText();
-                    if (collapsedText != null) {
-                        descriptors.add(new FoldingDescriptor(root.getNode(), textRange, null, collapsedText));
-                    } else {
-                        descriptors.add(new FoldingDescriptor(root.getNode(), textRange));
-                    }
+                    descriptors.add(new FoldingDescriptor(

```

Comment:
No need for the branching logic anymore as we're calling a `FoldingDescriptor` constructor signature that allows a `null` value for `collapsedText`. We also pass a flag denoting whether or not the region should be collapsed by default which is, of course, the whole point of the changes in this PR.